phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexTool.java

```diff
@@ -248,12 +247,13 @@ private CommandLine parseOptions(String[] args) {
                     + PARTIAL_REBUILD_OPTION.getLongOpt());
         }
         		
-        if (!(cmdLine.hasOption(DIRECT_API_OPTION.getOpt())) && cmdLine.hasOption(INDEX_TABLE_OPTION.getOpt())
+        if (!cmdLine.hasOption(INDEX_TABLE_OPTION.getOpt())
```

Comment:
This also looks incorrect. Is this check needed anymore?

Reply:
If there is no index table parameter supplied, what will runfg do? I think this check is needed.

Reply:
Shouldn't the check be that if there's no index table set and no partial build set, then something is wrong? 

What prevents a partial build's job from being run in the foreground (other than that the logic currently has an isDirect flag, which might also need to be removed?)

Reply:
In the original code, there is no PARTIAL_REBUILD mentioned in this if block.
I just did conversion of previous check like this: !(cmdLine.hasOption(DIRECT_API_OPTION.getOpt()) && cmdLine.hasOption(INDEX_TABLE_OPTION.getOpt()) ==> !(true && cmdLine.hasOption(INDEX_TABLE_OPTION.getOpt()) ==> !cmdLine.hasOption(INDEX_TABLE_OPTION.getOpt()

Reply:
@gjacoby126 Are you asking for more validations to be added here? Gokcen's changes seem to be simply replacing cmdLine.hasOption(INDEX_TABLE_OPTION.getOpt() with true.

Reply:
@priyankporwal -  Keep in mind that just because we're retiring the -direct flag, not _all_ runs will go through the direct paths now, because partial build exists. Collapsing it down to "true" isn't _quite_ right, and the check as it existed was already kind of weird. 

Most index tool invocations need to specify an index, except for partial build, which apparently will look up all indexes of a base table. Leaving out the index name and not invoking partial build should throw an error. Foreground or not shouldn't affect it. 

There used to be a check much later that only allowed for running in the foreground if -direct was selected, but I don't see a reason you couldn't run the partial build job in the foreground.

Reply:
We should probably also look to see if the partial build path still makes sense, and retire it if not, but that can potentially be a separate JIRA.

Reply:
@gjacoby126 Removed the check, thanks for explanation. Can you please re-review?

Reply:
@gokceni - don't we still need a check for "no index table set, no partial build set"? Other than that I think is ready to go.